apps/playground-web/src/components/blockchain-api/watch-event-preview.tsx

```diff
@@ -0,0 +1,61 @@
+"use client";
+
+import { THIRDWEB_CLIENT } from "@/lib/client";
+import { useEffect, useState } from "react";
+import { getContract, toTokens } from "thirdweb";
+import { base } from "thirdweb/chains";
+import { useContractEvents } from "thirdweb/react";
+import { transferEvent } from "thirdweb/extensions/erc20";
+
+type Item = {
+  from: string;
+  to: string;
+  value: string;
+};
+
+const shortenAddress = (address: string, length = 2) => {
+  return `${address.slice(0, length + 2)}...${address.slice(-length)}`;
+};
+
+const usdcContractOnBase = getContract({
+  address: "0x833589fCD6eDb6E08f4c7C32D4f71b54bdA02913",
+  chain: base,
+  client: THIRDWEB_CLIENT,
+});
+
+export function WatchEventPreview() {
+  const [items, setItems] = useState<Item[]>([]);
+  const contractEvents = useContractEvents({
+    contract: usdcContractOnBase,
+    events: [transferEvent()],
+    blockRange: 5,
+  });
+
+  useEffect(() => {
```

Comment:
no need to use a state/useEffect for this. You can just have it in a const directly or with a useMemo().

```suggestion
const items = useMemo(() => {
    if (!contractEvents.data?.length) return [];
    return contractEvents.data
      .map((item) => {
        const { from, to, value } = item.args;
        return {
          from: shortenAddress(from),
          to: shortenAddress(to),
          value: Number(toTokens(value, 6)).toFixed(1),
        };
      })
      .slice(-5);
  }, [contractEvents.data]);
```